<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>759</width>
    <height>537</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>ZQLoader</string>
  </property>
  <property name="sizeGripEnabled">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout_5">
   <item row="3" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Close|QDialogButtonBox::RestoreDefaults</set>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QProgressBar" name="progressBar">
     <property name="value">
      <number>0</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="textVisible">
      <bool>true</bool>
     </property>
     <property name="invertedAppearance">
      <bool>false</bool>
     </property>
     <property name="textDirection">
      <enum>QProgressBar::TopToBottom</enum>
     </property>
     <property name="format">
      <string>%p%</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QPushButton" name="pushButtonGo">
     <property name="text">
      <string>Go!</string>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Main Page</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <property name="spacing">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QScrollArea" name="scrollArea">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>733</width>
            <height>398</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <property name="leftMargin">
            <number>6</number>
           </property>
           <property name="topMargin">
            <number>6</number>
           </property>
           <property name="rightMargin">
            <number>6</number>
           </property>
           <property name="bottomMargin">
            <number>6</number>
           </property>
           <property name="spacing">
            <number>4</number>
           </property>
           <item row="1" column="9">
            <widget class="QPushButton" name="pushButtonBrowseTurboFile">
             <property name="text">
              <string>Browse...</string>
             </property>
            </widget>
           </item>
           <item row="3" column="3" alignment="Qt::AlignVCenter">
            <widget class="Dial" name="dialVolumeLeft">
             <property name="maximumSize">
              <size>
               <width>50</width>
               <height>50</height>
              </size>
             </property>
             <property name="minimum">
              <number>-100</number>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="wrapping">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="9">
            <widget class="QPushButton" name="pushButtonBrowseNormalFile">
             <property name="text">
              <string>Browse...</string>
             </property>
            </widget>
           </item>
           <item row="13" column="1" colspan="9">
            <widget class="QTextEdit" name="textEditOutput">
             <property name="sizeAdjustPolicy">
              <enum>QAbstractScrollArea::AdjustToContents</enum>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="html">
              <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;ZQLoader version 2.0&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Copyright (c) 2024 - Oxidaan&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;This project uses the MIT license. See LICENSE.txt for details.&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
            </widget>
           </item>
           <item row="3" column="9">
            <widget class="QPushButton" name="pushButtonTune">
             <property name="text">
              <string>Tune...</string>
             </property>
            </widget>
           </item>
           <item row="11" column="1" colspan="9">
            <widget class="Line" name="line_4">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QLabel" name="label_10">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;To write an output fil instead of playing sound enter .wav or .tzx output file here. Leave emtpy for normal usage (playing audio)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>Output file:</string>
             </property>
            </widget>
           </item>
           <item row="7" column="2" colspan="6">
            <widget class="QLineEdit" name="lineEditOutputFile"/>
           </item>
           <item row="4" column="1" colspan="9">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;Volume. Use negative values to invert channel. Use the 'Tune' button for an endless leader tone. Then adjust volume and cables until you see red/cyan stripes.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="textFormat">
              <enum>Qt::RichText</enum>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="13" column="9" alignment="Qt::AlignBottom">
            <widget class="QPushButton" name="pushButtonClean">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Clean</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Volume Left:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1" colspan="9">
            <widget class="Line" name="line">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="6" column="1" colspan="9">
            <widget class="Line" name="line_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Normal speed file:</string>
             </property>
            </widget>
           </item>
           <item row="10" column="1" colspan="9">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;To write an output file instead of playing sound enter a .wav or .tzx output file here. Leave empty for normal usage (playing audio)&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="textFormat">
              <enum>Qt::RichText</enum>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="3" column="7">
            <widget class="Dial" name="dialVolumeRight">
             <property name="maximumSize">
              <size>
               <width>50</width>
               <height>50</height>
              </size>
             </property>
             <property name="minimum">
              <number>-100</number>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="wrapping">
              <bool>false</bool>
             </property>
             <property name="notchesVisible">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="3" column="5">
            <widget class="QLineEdit" name="lineEditVolumeRight"/>
           </item>
           <item row="3" column="4">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Volume Right:</string>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QLineEdit" name="lineEditVolumeLeft"/>
           </item>
           <item row="1" column="2" colspan="6">
            <widget class="QLineEdit" name="lineEditTurboFile">
             <property name="toolTip">
              <string>Select a file to speed load here, a game for example.</string>
             </property>
            </widget>
           </item>
           <item row="7" column="9">
            <widget class="QPushButton" name="pushButtonBrowseOutputFile">
             <property name="text">
              <string>Browse...</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Turbo speed file:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2" colspan="6">
            <widget class="QLineEdit" name="lineEditNormalFile">
             <property name="toolTip">
              <string>Enter a file here that will be loaded at normal speed. By default this is the ZQ Loader itself.</string>
             </property>
            </widget>
           </item>
           <item row="0" column="8" rowspan="2">
            <widget class="QPushButton" name="pushButtonPreLoad">
             <property name="toolTip">
              <string>Preload ZQ Loader to ZX Spectrum,so you have time to browse to select file to turbo-load.</string>
             </property>
             <property name="text">
              <string>PreLoad</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Loader Settings</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5322">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <property name="spacing">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QScrollArea" name="scrollArea_2">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents_2">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>508</width>
            <height>349</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout_421">
           <property name="leftMargin">
            <number>6</number>
           </property>
           <property name="topMargin">
            <number>6</number>
           </property>
           <property name="rightMargin">
            <number>6</number>
           </property>
           <property name="bottomMargin">
            <number>6</number>
           </property>
           <property name="spacing">
            <number>4</number>
           </property>
           <item row="0" column="3" colspan="2">
            <widget class="QLineEdit" name="lineEditSampleRate"/>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>Zero TStates:</string>
             </property>
            </widget>
           </item>
           <item row="3" column="3" colspan="2">
            <widget class="QLineEdit" name="lineEditOneTStates"/>
           </item>
           <item row="14" column="4">
            <widget class="QLineEdit" name="lineEditLoaderAddress"/>
           </item>
           <item row="5" column="1">
            <widget class="QLabel" name="label_14">
             <property name="text">
              <string>Bit One Threshold (0=default):</string>
             </property>
            </widget>
           </item>
           <item row="13" column="1">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Compression Type:</string>
             </property>
            </widget>
           </item>
           <item row="14" column="1">
            <widget class="QLabel" name="label_16">
             <property name="text">
              <string>When loading snapshot put loader code at:</string>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>One TStates:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="3" colspan="2">
            <widget class="QLineEdit" name="lineEditEndOfByteDelay"/>
           </item>
           <item row="2" column="3" colspan="2">
            <widget class="QLineEdit" name="lineEditZeroTStates"/>
           </item>
           <item row="16" column="1">
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="6" column="3" colspan="2">
            <widget class="QLineEdit" name="lineEditBitLoopMax"/>
           </item>
           <item row="5" column="3" colspan="2">
            <widget class="QLineEdit" name="lineEditBitOneThreshold"/>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>Audio Sample Rate (0=device default):</string>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>End of Byte Delay (TStates):</string>
             </property>
            </widget>
           </item>
           <item row="13" column="3" colspan="2">
            <widget class="QComboBox" name="comboBoxCompressionType">
             <property name="currentIndex">
              <number>0</number>
             </property>
             <item>
              <property name="text">
               <string>None</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>RLE</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Automatic</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="15" column="1">
            <widget class="QLabel" name="label_17">
             <property name="text">
              <string>When using screen hide loader with attributes:</string>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="15" column="3">
            <widget class="QCheckBox" name="checkBoxFunAttribs">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="14" column="3">
            <widget class="QComboBox" name="comboBoxLoaderLocation">
             <item>
              <property name="text">
               <string>Screen (bottom 3rd)</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>In empty location in snapshot (automatic)</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>This address:</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="9" column="1">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>ZX Spectrum Clock Frequency (Hz):</string>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QLabel" name="label_15">
             <property name="text">
              <string>Bit Loop Max (0=default):</string>
             </property>
            </widget>
           </item>
           <item row="12" column="1" colspan="4">
            <widget class="Line" name="line_6">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="1" column="1" colspan="4">
            <widget class="Line" name="line_5">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="9" column="3" colspan="2">
            <widget class="QLineEdit" name="lineEditClock"/>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>About</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_532">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <property name="horizontalSpacing">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QScrollArea" name="scrollArea_3">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents_3">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>98</width>
            <height>700</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <property name="leftMargin">
            <number>6</number>
           </property>
           <property name="topMargin">
            <number>6</number>
           </property>
           <property name="rightMargin">
            <number>6</number>
           </property>
           <property name="bottomMargin">
            <number>6</number>
           </property>
           <property name="spacing">
            <number>4</number>
           </property>
           <item row="1" column="1">
            <widget class="QLabel" name="labelAbout">
             <property name="text">
              <string>&lt;html&gt;&lt;head/&gt;
&lt;body&gt;
&lt;p&gt;&lt;span style=&quot; font-family:'Segoe UI'; font-size:9pt;&quot;&gt;ZQLoader version 2.0&lt;/span&gt;&lt;/p&gt;
&lt;p&gt;&lt;span style=&quot; font-family:'Segoe UI'; font-size:9pt;&quot;&gt;Copyright (c) 2024 Daan Scherft [Oxidaan]&lt;/span&gt;&lt;/p&gt;
&lt;p&gt;&lt;span style=&quot; font-family:'Segoe UI'; font-size:9pt;&quot;&gt;This project uses the MIT license. See LICENSE.txt for details.&lt;/span&gt;&lt;/p&gt;
&lt;p&gt;&lt;span style=&quot; font-family:'Segoe UI'; font-size:9pt;&quot;&gt;See: &lt;/span&gt;&lt;a href=&quot;https://github.com/oxidaan/zqloader&quot;&gt;&lt;span style=&quot; font-family:'Segoe UI'; font-size:9pt; text-decoration: underline; color:#0000ff;&quot;&gt;oxidaan/zqloader: A quick loader for the ZX Spectrum&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;
&lt;p&gt;&lt;br/&gt;&lt;span style=&quot; font-family:'Segoe UI'; font-size:9pt;&quot;&gt;This project uses the miniaudio library by David Read. (See: &lt;/span&gt;&lt;a href=&quot;https://miniaud.io/&quot;&gt;&lt;span style=&quot; font-family:'Segoe UI'; font-size:9pt; text-decoration: underline; color:#0000ff;&quot;&gt;miniaudio - A single file audio playback and capture library.&lt;/span&gt;&lt;/a&gt;&lt;span style=&quot; font-family:'Segoe UI'; font-size:9pt;&quot;&gt;)&lt;/span&gt;&lt;/p&gt;
&lt;p&gt;&lt;br/&gt;&lt;span style=&quot; font-family:'Segoe UI'; font-size:9pt;&quot;&gt;This project uses QT. (v6.4)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Segoe UI'; font-size:9pt;&quot;&gt;See: &lt;/span&gt;&lt;a href=&quot;https://www.qt.io/&quot;&gt;&lt;span style=&quot; font-family:'Segoe UI'; font-size:9pt; text-decoration: underline; color:#0000ff;&quot;&gt;Qt | Tools for Each Stage of Software Development Lifecycle&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QPushButton" name="pushButtonTest">
             <property name="text">
              <string>Debug/Test</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Dial</class>
   <extends>QDial</extends>
   <header>dial.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>lineEditNormalFile</tabstop>
  <tabstop>pushButtonBrowseNormalFile</tabstop>
  <tabstop>lineEditTurboFile</tabstop>
  <tabstop>pushButtonBrowseTurboFile</tabstop>
  <tabstop>lineEditVolumeLeft</tabstop>
  <tabstop>dialVolumeLeft</tabstop>
  <tabstop>lineEditVolumeRight</tabstop>
  <tabstop>dialVolumeRight</tabstop>
  <tabstop>pushButtonTune</tabstop>
  <tabstop>lineEditOutputFile</tabstop>
  <tabstop>pushButtonBrowseOutputFile</tabstop>
  <tabstop>pushButtonGo</tabstop>
  <tabstop>pushButtonClean</tabstop>
  <tabstop>scrollArea</tabstop>
  <tabstop>lineEditSampleRate</tabstop>
  <tabstop>lineEditBitLoopMax</tabstop>
  <tabstop>scrollArea_2</tabstop>
  <tabstop>lineEditOneTStates</tabstop>
  <tabstop>lineEditZeroTStates</tabstop>
  <tabstop>lineEditBitOneThreshold</tabstop>
  <tabstop>textEditOutput</tabstop>
  <tabstop>tabWidget</tabstop>
  <tabstop>lineEditEndOfByteDelay</tabstop>
  <tabstop>checkBoxFunAttribs</tabstop>
  <tabstop>lineEditLoaderAddress</tabstop>
  <tabstop>comboBoxCompressionType</tabstop>
  <tabstop>comboBoxLoaderLocation</tabstop>
  <tabstop>scrollArea_3</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
